apps/05Mastermind.js

```diff
@@ -26,12 +26,63 @@ function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
+function generateHint(solution, guess) {
     // your code here
+    var solutionArray = solution.split("");
+    var guessArray = guess.split("");
+    var correctLetterLocations = 0;
+    var correctLetters = 0;
+
+    for (var i = 0; i < solutionArray.length; i++) {
+      if (solutionArray[i] === guessArray[i]) {
+        correctLetterLocations++;
+        solutionArray[i] = null;
+      }
+    }
+
+    for (var i = 0; i < solutionArray.length; i++) {
+      var targetIndex = solutionArray.indexOf(guessArray[i]);
+      if (targetIndex > -1) {
+        correctLetters++;
+        solutionArray[targetIndex] = null;
+      }
+    }
+    // return colors.red(correctLetterLocations) + "-" + colors.white(correctLetters);
+    return correctLetterLocations + "-" + correctLetters;
+}
+
+function addColor(hint) {
+    var hintNumb = hint.split('-');
+    return colors.red(hintNumb[0]) + "-" + colors.white(hintNumb[1]);
+}
+
+function gameOVERMAN(){
+  board = [];
+  solution = '';
+  generateSolution();
 }
 
 function mastermind(guess) {
     // your code here
+    // solution = "abcd";
+
+
+    if (guess === solution) {
+      console.log("You guessed it!");
+      gameOVERMAN();
+      return 'Want to play again?'
```

Comment:
It looks like you're failing a test.  All your stuff works, it's just that your game reset is causing one of the tests to fail.

This test right here:

```javascript
        it('should be able to detect a win', function () {
            assert.equal(mastermind(solution), 'You guessed it!');
        });
```

See the problem.